webclient/pas/views.py

```diff
@@ -711,22 +729,31 @@ def smazat(request, ident_cely):
         )
     if request.method == "POST":
         nalez.deleted_by_user = request.user
-        nalez.create_transaction(request.user, ZAZNAM_USPESNE_SMAZAN, ZAZNAM_SE_NEPOVEDLO_SMAZAT)
-        nalez.close_active_transaction_when_finished = True
-        nalez.record_deletion(nalez.active_transaction)
-        resp1 = nalez.delete()
-        if resp1:
-            logger.info(
-                "pas.views.smazat.deleted",
-                extra={"value": resp1},
-            )
-            return JsonResponse({"redirect": reverse("pas:index")})
-        else:
-            logger.warning("pas.views.smazat.not_deleted", extra={"ident_cely": ident_cely})
-            return JsonResponse(
-                {"redirect": reverse("pas:detail", kwargs={"ident_cely": ident_cely})},
-                status=403,
-            )
+        fedora_transaction = nalez.create_transaction(request.user, ZAZNAM_USPESNE_SMAZAN, ZAZNAM_SE_NEPOVEDLO_SMAZAT)
+        try:
+            nalez.igsn_delete()
+            nalez.close_active_transaction_when_finished = True
+            nalez.record_deletion(nalez.active_transaction)
+            resp1 = nalez.delete()
+            if resp1:
+                logger.info(
+                    "pas.views.smazat.deleted",
+                    extra={"value": resp1},
+                )
+                return JsonResponse({"redirect": reverse("pas:index")})
+            else:
```

Comment:
Kdy toto vlastně nastane? Neměly by se ošetřit rollbacky?

Reply:
Teoreticky nastane v případě, že nedojde ke smazání, ale současně pokus o mazání nezpůsobí chybu. V reálu a v této konkrétní situaci přesně nevím, co by to mohlo způsobit.